### ISSUE TITLE
Onigashima .lua script will not run

### ISSUE THREAD

**Issue by kborb** [sudgy/nintendo_translations#1]
(I have basically 0 coding experience so I apologize for my potential ineptitude here)

Hello! I recently found this project and was interested in trying it out with Shin Onigashima, but I've been having trouble getting the .lua file for the game working. As far as I'm aware I've properly followed every step of the guide, but when I run the .lua script the Windows command prompt opens and closes immediately, and nothing else happens after that. I'm running the game on Mesen 0.9.9, and I have everything set up in the same directory, including the emulator and the rom. 

I'm not sure what I'm doing wrong here, so your help would be very much appreciated. Thank you in advance, and sorry to bother you!

**Comment by sudgy** [comment]
Sorry that I missed this for so long!  It sounds like you are trying to run the script from outside the emulator.  The command prompt shouldn't ever open.  The emulator should have an option for Lua scripts somewhere, and you need to open the Lua script in the emulator.  I seem to remember it was different on Windows than on Linux and I don't have a Windows machine available to me at the moment so I don't remember exactly where it is, but there should be some menu option that says "Script Window" or something.

**Comment by kborb** [comment]
Oh that's exactly where it was! I had no idea you could open scripts like that on Mesen haha.
However, when I tried to run the script it gave me an "invalid escape sequence" error on the first line, where you paste the directory.

**Comment by sudgy** [comment]
Did you use backslashes in the directory?  I think you need to use forward slashes.

**Comment by kborb** [comment]
I did, and that seemed to be the issue there, although I also had to add a slash in front of mesen.lua on the next line so it would stop reading the file name as part of the directory lol (I don't know if this actually fixed that issue but it seemed to stop the error at least). But now when I try to run the script it shows me this:
![image_2024-08-31_204654108](https://github.com/user-attachments/assets/73bb0aa0-5dd9-4fe5-ae7c-a922961d893e)

**Comment by sudgy** [comment]
Are you sure it's not working?  The script doesn't kick in until a new message appears, and I left the title screen untranslated.  Try starting the script on the title screen and then pressing start to see if the message there is still in Japanese.

**Comment by kborb** [comment]
I just tried that now and yeah it's still in Japanese. The text you see when you sit on the title screen for long enough isn't changed either, if that was also translated.

**Comment by sudgy** [comment]
Oh looking into this a bit more it seems like it can't open the file "onigashima_messages.bin" for some reason.  You can try adding `e.log(messages_filename)` right before line 58 to see what file the script is trying to open.

...also, I just now realized I accidentally put "rb" as an argument to `assert`, not `io.open`.  That might have actually been the cause of a bunch of issues I've had in the past.  You might get a more helpful error message if on line 58 you move the `)` after `messages_filename` to the end of the line next to the other `)`.  I'll fix this myself at some point soon, but I need to test that it won't break anything first.

**Comment by kborb** [comment]
I followed what you said and this is the error message that I got (the blank spots in the directory are my actual name lol):
![image_2024-09-03_095435109](https://github.com/user-attachments/assets/5dda0a8d-dd3f-4444-9291-d5cd7c6efb44)

Since this was an issue I was having before (with it combining mesen.lua into the directory improperly) i added forward slashes to the messages and options .bin files, and it actually started working from this! I've only skimmed through the game a little bit so far but the script seems to be running fine. I don't know how helpful that would be for you though haha.
